File: adoptame-api/src/main/java/mx/edu/unpa/app_pet/domains/Usuario.java
package mx.edu.unpa.app_pet.domains;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "Usuario")
@Data
@NoArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "idUsuario")
    private Integer idUsuario;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(name = "apellidoPaterno", nullable = false, length = 100)
    private String apellidoPaterno;

    @Column(name = "apellidoMaterno", length = 100)
    private String apellidoMaterno;

    @Column(nullable = false, length = 150, unique = true)
    private String email;

    @Column(nullable = false, length = 255)
    private String password;

    @Column(length = 20)
    private String telefono;

    private Boolean activo = true;

    @Column(name = "fechaRegistro", insertable = false, updatable = false)
    private LocalDateTime fechaRegistro;
}

File: adoptame-app/app/src/main/java/mx/edu/unpa/adoptame/model/Usuario.kt
package mx.edu.unpa.adoptame.model

import com.google.gson.annotations.SerializedName

data class Usuario(
    @SerializedName("idUsuario")        val idUsuario: Int,
    @SerializedName("nombre")           val nombre: String,
    @SerializedName("apellidoPaterno")  val apellidoPaterno: String,
    @SerializedName("apellidoMaterno")  val apellidoMaterno: String?,
    @SerializedName("email")            val email: String,
    @SerializedName("telefono")         val telefono: String?,
    @SerializedName("activo")           val activo: Boolean,
    @SerializedName("fechaRegistro")    val fechaRegistro: String
) {
    val nombreCompleto: String
        get() = "$nombre $apellidoPaterno ${apellidoMaterno.orEmpty()}".trim()
}